lab/skel_graph/traversal/problems/alee/alee.cpp

```diff
@@ -0,0 +1,152 @@
+			std::cin >> x >> y;
+			graph[x][y] = 'T';
+		}
+
+        // Se citesc coordonatele portii din care se porneste cautarea, 
+        // si cele ale portii ce reprezinta destinatia
+        std::cin >> startGate.first >> startGate.second 
+            >> endGate.first >> endGate.second;
+	}
+	
+    // intrucat graful este reprezentat sub forma unei matrice
+    // se va aplica Algoritmul lui Lee
+	int get_result() {
+        // se tine evidenta nodurilor vizitate, 
+        // initial toate nodurile sunt nevizitate
+		std::vector<std::vector<bool>> visited(n + 1, 
+            std::vector<bool>(n + 1, false));
+
+        // vectorul de distante pornind de la poarta de start
+		std::vector<std::vector<int>> minPath(n + 1, 
+            std::vector<int>(n + 1, INT_MAX));
+
+        // in coada se vor retine nodurile descoperite, dar inca nefinalizate
+		std::queue<std::pair<int, int>> nodeQueue;
+
+        // vectorul ce reprezinta posibilele pozitii 
+        // ale vecinilor unui element din matrice
+        std::vector<std::pair<int, int>> directions;
+        directions.push_back({1, 0});
+        directions.push_back({0, 1});
+        directions.push_back({-1, 0});
+        directions.push_back({0, -1});
+
+        // initializari pentru poarta de start
+		visited[startGate.first][startGate.second] = true;
+		minPath[startGate.first][startGate.second] = 1;
+		nodeQueue.push({startGate.first, startGate.second});
+
+		while (nodeQueue.size()) {
+            // se extrage primul element din coada
+			auto node = nodeQueue.front();
+
+            // se parcurg vecinii acestuia folosind vectorul de directii
+            // acestia pot fi maximum patru
+			for (auto& dir : directions) {
+                // se verifica daca se respecta granitele matricei
+                if (correctBounds(node, dir)) {
+                    int x = node.first + dir.first;
+                    int y = node.second + dir.second;
+                    if (!visited[x][y]) {
+                        minPath[x][y] = minPath[node.first][node.second] + 1;
+                        visited[x][y] = true;
+
+                        // elementul curent este bagat in coada
+                        // doar daca reprezinta o zona libera
+                        // un copac ar impiedica inaintarea pe un drum
+                        if (graph[x][y] == ' ') {
+                            nodeQueue.push({x, y});
+                        }
+                    }
+                }
+			}
+            // se elimina din coada
+			nodeQueue.pop();
+        }
+
+        // returneaza distanta de la poarta de start la poarta destinatie
+		return minPath[endGate.first][endGate.second];
+	}
+	
+    // verifica daca coordonatele unui vecin sunt valide 
+    // adica nu depasesc extremele matricei
+    bool correctBounds(const std::pair<int, int>& p1, const std::pair<int, int>& p2) {
+        if (p1.first + p2.first <= n && p1.first + p2.first >= 1 
```

Comment:
Maybe here we can directly return the result? I mean return (condition1 && condition2 && condition3 && condition4) ?